TECHMANIA/Assets/Scripts/UniversalIO.cs

```diff
@@ -45,4 +149,40 @@ public static string ReadAllText(string path)
             return File.ReadAllText(path);
         }
     }
+
+    public static void WriteAllText(string path, string content)
+    {
+        if (path.StartsWith(ANDROID_CONTENT_URI))
+        {
+            AndroidNativeIO.IO.File.WriteAllText(path, content);
+        } else
+        {
+            File.WriteAllText(path, content);
+        }
+    }
+
+    #endregion
+
+    #region Path
+
+    public static string PathGetDirectoryName(string path, bool returnToTreeUri = true)
+    {
+
+        if (path.StartsWith(ANDROID_CONTENT_URI))
+        {
+            return AndroidNativeIO.IO.Path.GetDirectoryName(path, returnToTreeUri);
+        }
+        return Path.GetDirectoryName(path);
+    }
+
+    public static string PathCombine(params string[] path)
+    {
+        if (path[0].StartsWith(ANDROID_CONTENT_URI))
+        {
+            return AndroidNativeIO.IO.Path.Combine(path[0], path[1]);
```

Comment:
What if there are more than 2 parameters?

If `AndroidNativeIO.IO.Path.Combine` doesn't accept variable numbers of arguments, then maybe create overloads of `PathCombine` with 2, 3, and/or more parameters.